src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IVariableDeclaration.vb

```diff
@@ -0,0 +1,1167 @@
+
+            Assert.Equal(2, operation.Variables.Count())
+            Dim var1 = operation.Variables(0)
+            Dim var2 = operation.Variables(1)
+
+            Assert.NotNull(var1.InitialValue)
+            Assert.NotNull(var2.InitialValue)
+
+            Assert.Same(var1.InitialValue, var2.InitialValue)
+        End Sub
+
+        <Fact(), WorkItem(17599, "https://github.com/dotnet/roslyn/issues/17599")>
+        Public Sub ArrayDeclarationWithLength()
+            Dim source = <![CDATA[
+Module Program
+    Sub Main(args As String())
+        Dim i1(2) As Integer'BIND:"Dim i1(2) As Integer"
+    End Sub
+End Module
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IVariableDeclarationStatement (1 variables) (OperationKind.VariableDeclarationStatement)
+  IVariableDeclaration: i1 As System.Int32() (OperationKind.VariableDeclaration)
+    Initializer: IArrayCreationExpression (Dimension sizes: 1, Element Type: System.Int32) (OperationKind.ArrayCreationExpression, Type: System.Int32())
+        IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32, Constant: 3)
+          Left: ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+          Right: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of LocalDeclarationStatementSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact(), WorkItem(17599, "https://github.com/dotnet/roslyn/issues/17599")>
+        Public Sub ArrayDeclarationMultipleVariables()
+            Dim source = <![CDATA[
+Module Program
+    Sub Main(args As String())
+        Dim i1(), i2 As Integer'BIND:"Dim i1(), i2 As Integer"
+
+    End Sub
+End Module
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IVariableDeclarationStatement (2 variables) (OperationKind.VariableDeclarationStatement)
+  IVariableDeclaration: i1 As System.Int32() (OperationKind.VariableDeclaration)
+  IVariableDeclaration: i2 As System.Int32 (OperationKind.VariableDeclaration)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of LocalDeclarationStatementSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact(), WorkItem(17599, "https://github.com/dotnet/roslyn/issues/17599")>
+        Public Sub ArrayDeclarationInvalidAsNew()
+            Dim source = <![CDATA[
+Module Program
+    Sub Main(args As String())
+        Dim i1(2) As New Integer'BIND:"Dim i1(2) As New Integer"
+    End Sub
+End Module
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IVariableDeclarationStatement (1 variables) (OperationKind.VariableDeclarationStatement, IsInvalid)
+  IVariableDeclaration: i1 As System.Int32() (OperationKind.VariableDeclaration, IsInvalid)
+    Initializer: IInvalidExpression (OperationKind.InvalidExpression, Type: System.Int32(), IsInvalid)
```

Comment:
>Initializer: IInvalidExpression (OperationKind.InvalidExpression, Type: System.Int32(), IsInvalid) [](start = 4, length = 98)

This looks like another scenario where IInvalidExpression hides nodes - the array size in this case.